go/vt/schemadiff/schema.go

```diff
@@ -763,15 +761,130 @@ func (s *Schema) apply(diffs []EntityDiff) error {
 // The operation does not modify this object. Instead, if successful, a new (modified) Schema is returned.
 func (s *Schema) Apply(diffs []EntityDiff) (*Schema, error) {
 	dup := s.copy()
-	for k, v := range s.named {
-		dup.named[k] = v
-	}
 	if err := dup.apply(diffs); err != nil {
 		return nil, err
 	}
 	return dup, nil
 }
 
+// SchemaDiff calulates a rich diff between this schema and the given schema. It is stronger than Diff().
+// On top of returning the list of diffs that can take this schema into the given schema, this function also
+// evaluates the dependencies between those diffs, if any.
+func (s *Schema) SchemaDiff(other *Schema, hints *DiffHints) (*SchemaDiff, error) {
```

Comment:
Do we want to deprecate / remove `Diff`? I guess it uses it internally, but should we then rename `Diff` to `diff` to not export it?

Reply:
Yes, I intend to do that, but in a separate PR, to have a grace period of a couple weeks where we can use either.